mesheryctl/pkg/utils/relationship.go

```diff
@@ -0,0 +1,224 @@
+				}
+			}
+
+			if len(versions) == 0 {
+				continue
+			}
+		} else {
+			versions = []string{relationship.Version}
+		}
+
+		for _, version := range versions {
+			if relationship.Model == "" {
+				continue
+			}
+			var rel _rel.RelationshipDefinition
+			rel.SchemaVersion = v1alpha3.RelationshipSchemaVersion
+			rel.Kind = _rel.RelationshipDefinitionKind(utils.ReplaceSpacesAndConvertToLowercase(relationship.KIND))
+			rel.RelationshipType = utils.ReplaceSpacesAndConvertToLowercase(relationship.Type)
+			rel.SubType = utils.ReplaceSpacesAndConvertToLowercase(relationship.SubType)
+			rel.EvaluationQuery = &relationship.EvalPolicy
+
+			rel.Version = "v1.0.0"
+			rel.Model = model.ModelDefinition{
+				Name:  relationship.Model,
+				Model: model.Model{Version: version},
+			}
+			status := _rel.Enabled
+			rel.Status = &status
+			var styles _rel.RelationshipDefinitionMetadataStyles
+
+			if relationship.Styles != "" {
+				err := encoding.Unmarshal([]byte(relationship.Styles), &styles)
+				if err != nil {
+					Log.Error(err)
+					continue
+				}
+				if rel.Metadata == nil {
+					rel.Metadata = &_rel.Relationship_Metadata{}
+				}
+			}
+
+			rel.Metadata = &_rel.Relationship_Metadata{
+				Description: &relationship.Description,
+				Styles:      &styles,
+			}
+
+			var selectorSet _rel.SelectorSet
+			if relationship.Selector != "" {
+				err := encoding.Unmarshal([]byte(relationship.Selector), &selectorSet)
+				if err != nil {
+					Log.Error(err)
+					continue
+				}
+				rel.Selectors = &selectorSet
+			}
+
+			fullPath, err := ConstructRelationshipPath(relationship.Model, version, rel.Version, "../server/meshmodel", rel)
+			if err != nil {
+				Log.Error(err)
+				continue
+			}
+
+			err = utils.WriteJSONToFile(fullPath, rel)
+			if err != nil {
+				Log.Error(err)
+				continue
+			}
+
+			Log.Info(fmt.Sprintf("Relationship written to %s", fullPath))
+		}
+	}
+}
+
+func ConstructRelationshipPath(modelName, version, defVersion, outputLocation string, rel _rel.RelationshipDefinition) (string, error) {
+	// Construct the base path using model and version
+	basePath := fmt.Sprintf("%s/%s/%s/v1.0.0", outputLocation, modelName, version)
+	if _, err := os.Stat(basePath); os.IsNotExist(err) {
+		return "", err
+	}
+
+	var fromKind, toKind string
+
+	if rel.Selectors != nil && len(*rel.Selectors) > 0 {
+		firstSelector := (*rel.Selectors)[0]
+		if len(firstSelector.Allow.From) > 0 && len(firstSelector.Allow.To) > 0 {
+			if firstSelector.Allow.From[0].Kind != nil {
+				fromKind = *firstSelector.Allow.From[0].Kind
+			}
+			if firstSelector.Allow.To[0].Kind != nil {
+				toKind = *firstSelector.Allow.To[0].Kind
+			}
+		}
+	}
+	// Construct the filename
+	//filename:- kind-type-subtype-from-{selector from kind}-to-{selector to kind}.json
+	filename := fmt.Sprintf("%s-%s-%s-from-%s-to-%s.json",
+		rel.Kind, rel.RelationshipType, rel.SubType,
+		utils.ReplaceSpacesAndConvertToLowercase(fromKind),
```

Comment:
Any specific reason? These kinds maps directly to "component.component.kind", and all of the consumers of  relationships performs the exact match.

If we are changing it ensure components defs and all other places are also updated to reflect this change.

Reply:
It was just for naming convection when was naming the files after splitting lee mentioned to use lowercase.